src/advanced_inv.cpp

```diff
@@ -1036,41 +1066,70 @@ void advanced_inventory::redraw_pane( side p )
 {
     auto &spane = panes[src];
     auto &dpane = panes[dest];
 
     // AIM_ALL source area routine
     if(spane.get_area() == AIM_ALL) {
-        // future feature :-)
-        popup(_("You can't do that (yet!)"));
-        return false;
-        // make a copy of the current pane for below loop
+        bool done = false;
+        // copy the current pane, to be restored after the move is queued
         auto shadow = panes[src];
         // here we recursively call this function with each area in order to 
         // put all items in the proper destination area, with minimal fuss
         auto &loc = uistate.adv_inv_aim_all_location;
-        // vehicle items for said square
-        if(squares[loc].can_store_in_vehicle()) {
-            // either do the inverse of the pane (if it is the one we are transferring to),
-            // or just transfer the contents (if it is not the one we are transferring to)
-            spane.set_area(squares[loc], (dpane.get_area() == loc) ? !dpane.in_vehicle() : true);
-            // add items, calculate weights and volumes... the fun stuff
-            recalc_pane(src);
-            // then move the items to the destination area
-            move_all_items();
-        }
-        // same as above, but for map items
-        spane.set_area(squares[loc++], false);
-        recalc_pane(src);
-        move_all_items();
-        if(loc > AIM_NORTHEAST) {
-            loc = 0;
+        // re-entry nonsense
+        auto &entry = uistate.adv_inv_re_enter_move_all;
+        // if we are just starting out, set entry to initial value
+        switch(static_cast<aim_entry>(entry++)) {
+            case ENTRY_START:
+                ++entry;
+            case ENTRY_VEHICLE:
+                if(squares[loc].can_store_in_vehicle()) {
+                    // either do the inverse of the pane (if it is the one we are transferring to),
+                    // or just transfer the contents (if it is not the one we are transferring to)
+                    spane.set_area(squares[loc], (dpane.get_area() == loc) ? !dpane.in_vehicle() : true);
+                    // add items, calculate weights and volumes... the fun stuff
+                    recalc_pane(src);
+                    // then move the items to the destination area
+                    move_all_items(true);
+                }
+                break;
+            case ENTRY_MAP:
+                spane.set_area(squares[loc++], false);
+                recalc_pane(src);
+                move_all_items(true);
+                break;
+            case ENTRY_RESET:
+                if(loc > AIM_NORTHEAST) {
+                    loc = done = true;
+                    entry = -1;
+                } else {
+                    entry = 0;
+                }
+                break;
+            default:
+                debugmsg("Invalid `aim_entry' [%d] reached!", entry - 1);
+                entry = -1;
+                loc = 1;
```

Comment:
If you're already introducing enum values, you should do so consistently: 1 is `AIM_SOUTHWEST`, 0 (for entry above) is `ENTRY_VEHICLE` and -1 is `ENTRY_START`

Btw. it may be useful to define some constants that mark the range of the 9 `aim_location`s from `AIM_SOUTHWEST` to `AIM_NORTHEAST`. That range is used at several places and it would be more obvious what is meant by referring to `AROUND_LOC_FIRST`, `AROUND_LOC_LAST` (better name needed).